tests/rptest/tests/audit_log_test.py

```diff
@@ -1924,84 +1993,44 @@ def test_sr_audit(self):
                          auth=(self.username, self.password))
         assert r.status_code == requests.codes.ok
 
-        def match_api_user(endpoint, user, svc_name, record):
-            if record['class_uid'] == 6003 and record['dst_endpoint'][
-                    'svc_name'] == svc_name:
-                regex = re.compile(
-                    "http:\/\/(?P<address>.*):(?P<port>\d+)\/(?P<handler>.*)")
-                url_string = record['http_request']['url']['url_string']
-                match = regex.match(url_string)
-                if match and match.group('handler') == endpoint and record[
-                        'actor']['user']['name'] == user:
-                    return True
-
-            return False
-
         records = self.find_matching_record(
-            lambda record: match_api_user("subjects", self.username, self.
-                                          sr_audit_svc_name, record),
+            lambda record: self.match_api_record(record, "subjects"),
             lambda record_count: record_count >= 1, 'sr get api call')
 
-        assert self.aggregate_count(
-            records
-        ) == 1, f'Expected one record found {self.aggregate_count(records)}: {records}'
-
-        def match_authn_user(user, svc_name, result, record):
-            return record['class_uid'] == 3002 and record['dst_endpoint'][
-                'svc_name'] == svc_name and record['user'][
-                    'name'] == user and record['status_id'] == result
+        assert self.aggregate_count(records) == 1, \
+            f'Expected one record found {self.aggregate_count(records)}: {records}'
 
         _ = self.find_matching_record(
-            lambda record: match_authn_user(self.username, self.
-                                            sr_audit_svc_name, 1, record),
+            lambda record: self.match_authn_record(record, StatusID.SUCCESS),
```

Comment:
question (not related to your change): any idea why we don't assert here? like what is this testing?

Reply:
The "assertion" here is implicit because `find_matching_record` would throw a `TimeoutError` if we don't find a matching record.